src/strategies/hats-protocol-single-vote-per-org/index.ts

```diff
@@ -0,0 +1,180 @@
+  switch (network) {
+    case '1':
+      result = await subgraphRequestHats(
+        'https://api.thegraph.com/subgraphs/name/hats-protocol/hats-v1-ethereum',
+        options.humanReadableTreeId
+      );
+    case '10':
+      result = await subgraphRequestHats(
+        'https://api.thegraph.com/subgraphs/name/hats-protocol/hats-v1-optimism',
+        options.humanReadableTreeId
+      );
+      break;
+    case '5':
+      result = await subgraphRequestHats(
+        'https://api.thegraph.com/subgraphs/name/hats-protocol/hats-v1-goerli',
+        options.humanReadableTreeId
+      );
+      break;
+    case '137':
+      result = await subgraphRequestHats(
+        'https://api.thegraph.com/subgraphs/name/hats-protocol/hats-v1-polygon',
+        options.humanReadableTreeId
+      );
+      break;
+    case '100':
+      result = await subgraphRequestHats(
+        'https://api.thegraph.com/subgraphs/name/hats-protocol/hats-v1-gnosis-chain',
+        options.humanReadableTreeId
+      );
+      break;
+    case '42161':
+      result = await subgraphRequestHats(
+        'https://api.thegraph.com/subgraphs/name/hats-protocol/hats-v1-arbitrum',
+        options.humanReadableTreeId
+      );
+      break;
+  }
+
+  let wearersInAddresses = <any>[];
+
+  addresses.forEach((address) => {
+    const wearer = checkIfExists(address, result.tree);
+    wearersInAddresses = wearersInAddresses.concat(wearer);
+  });
+
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+
+  wearersInAddresses.forEach((wearer) => {
+    multi.call(wearer.address, options.address, 'isWearerOfHat', [
+      wearer.address,
+      wearer.hat
+    ]);
+  });
+
+  const multiResult = await multi.execute();
+
+  const myObj = {};
+  wearersInAddresses.forEach((wearer) => {
+    myObj[wearer.address] = 0;
+    for (const result of multiResult) {
+      if (wearer.address === result.address) {
+        myObj[wearer.address] = 1;
+        break;
+      }
+    }
+  });
+
+  return myObj;
+}
+
+class Multicaller {
+  public network: string;
+  public provider: StaticJsonRpcProvider;
+  public abi: any[];
+  public options: any = {};
+  public calls: any[] = [];
+  public paths: any[] = [];
+
+  constructor(
+    network: string,
+    provider: StaticJsonRpcProvider,
+    abi: any[],
+    options?
+  ) {
+    this.network = network;
+    this.provider = provider;
+    this.abi = abi;
+    this.options = options || {};
+  }
+
+  call(path, address, fn, params?): Multicaller {
+    this.calls.push([address, fn, params]);
+    this.paths.push(path);
+    return this;
+  }
+
+  async execute(): Promise<any> {
+    const obj = <any>[];
+    const result = await multicall(
+      this.network,
+      this.provider,
+      this.abi,
+      this.calls,
+      this.options
+    );
+    result.forEach((r, i) => {
+      obj.push({
+        address: this.paths[i],
+        value: r
+      });
+    });
+    this.calls = [];
+    this.paths = [];
+    return obj;
+  }
+}
```

Comment:
```suggestion
```

Reply:
same thing here, I am using a modified version of the Multicaller class.

Reply:
@Hotmanics Main idea is to avoid duplicate code as much as possible. I see you change and feel like we can modify the same data after executing the multicaller

Reply:
@ChaituVR From my understanding, the execute() function from the utils.ts file flattens the values.
For instance, if I have several calls like so:
```
    multi.call('0xaddress1', 'SmartContractAddress', 'isWearerOfHat', ['0xaddres1' '0xHatId1']);
    multi.call('0xaddress1', 'SmartContractAddress', 'isWearerOfHat', ['0xaddres1' '0xHatId2']);
    multi.call('0xaddress2', 'SmartContractAddress', 'isWearerOfHat', ['0xaddres2' '0xHatId1']);
```
Then the returned value from 'execute' will be something like so:
```
    {
       '0xaddress1': true,
       '0xaddress2': true
   }

```
When the desired result is something like:
```
    {
       '0xaddress1': true,
       '0xaddress1': true,
       '0xaddress2': true
   }
```

Reply:
It return entire value https://github.com/snapshot-labs/snapshot.js/blob/master/src/utils/multicaller.ts#L40C54-L40C71 you can modify result after getting it

Reply:
> It return entire value https://github.com/snapshot-labs/snapshot.js/blob/master/src/utils/multicaller.ts#L40C54-L40C71 you can modify result after getting it

I don't believe so unless I'm looking at incorrectly.

here is what my Multicaller returns. You can see that it returns multiple entries for 0xa6aF0566EF4eF7E8f38913f69d4e55c06F00A5aC.
<img width="1235" alt="Screenshot 2023-07-20 at 4 28 03 PM" src="https://github.com/snapshot-labs/snapshot-strategies/assets/32080359/44a211ee-8254-4aa6-aa9d-2ae7d6eb8228">

here is what snapshot's Multicaller returns. You can see that it removes all extra entries pertaining to 0xa6aF0566EF4eF7E8f38913f69d4e55c06F00A5aC and gives one result.
<img width="1238" alt="Screenshot 2023-07-20 at 4 29 47 PM" src="https://github.com/snapshot-labs/snapshot-strategies/assets/32080359/73fc0dd3-a2c3-4a35-b625-12623a93d6ca">

Apologies if I am missing something here. This is just what I am seeing on my end.